csharp/src/Microsoft.ML.OnnxRuntime/OrtFloat16.shared.cs

```diff
@@ -0,0 +1,1350 @@
+
+        private const ushort PositiveInfinityBits = 0x7C00;
+        private const ushort NegativeInfinityBits = 0xFC00;
+
+        private const ushort PositiveQNaNBits = 0x7E00;
+        private const ushort NegativeQNaNBits = 0xFE00;
+
+        private const ushort MinValueBits = 0xFBFF;
+        private const ushort MaxValueBits = 0x7BFF;
+
+        private const ushort PositiveOneBits = 0x3C00;
+        private const ushort NegativeOneBits = 0xBC00;
+
+        private const ushort EBits = 0x4170;
+        private const ushort PiBits = 0x4248;
+        private const ushort TauBits = 0x4648;
+
+        // Well-defined and commonly used values
+
+        /// <summary>
+        /// Float16 Epsilon value
+        /// </summary>
+        public static Float16 Epsilon => new Float16(EpsilonBits);                        //  5.9604645E-08
+
+        /// <summary>
+        /// Float16 Pi value
+        /// </summary>
+        public static Float16 Pi => new Float16(PiBits);                                 //  3.14159265358979323846
+
+        /// <summary>
+        /// Float16 Positive Infinity value
+        /// </summary>
+        public static Float16 PositiveInfinity => new Float16(PositiveInfinityBits);      //  1.0 / 0.0;
+
+        /// <summary>
+        /// Float16 Negative Infinity value
+        /// </summary>
+        public static Float16 NegativeInfinity => new Float16(NegativeInfinityBits);      // -1.0 / 0.0
+
+        /// <summary>
+        /// Float16 NaN
+        /// </summary>
+        public static Float16 NaN => new Float16(NegativeQNaNBits);                       //  0.0 / 0.0
+
+        /// <summary>
+        /// Float16 Zero value
+        /// </summary>
+        public static Float16 Zero => new Float16(PositiveZeroBits);                      //  0.0
+
+        /// <summary>
+        /// Float16 One value
+        /// </summary>
+        public static Float16 One => new Float16(OneBits);                                //  1.0
+
+        /// <summary>
+        /// Float16 Negative Zero value
+        /// </summary>
+        public static Float16 NegativeZero => new Float16(NegativeZeroBits);              // -0.0
+
+        /// <summary>
+        /// Float16 Min value
+        /// </summary>
+        public static Float16 MinValue => new Float16(MinValueBits);                      // 64,511
+
+        /// <summary>
+        /// Float16 Max value
+        /// </summary>
+        public static Float16 MaxValue => new Float16(MaxValueBits);                      //  31,743
+
+        /// <summary>
+        /// float16 representation bits
+        /// </summary>
+        public readonly ushort value;
+
+        /// <summary>
+        /// Ctor from ushort bits, no conversion is done
+        /// </summary>
+        /// <param name="v"></param>
+        public Float16(ushort v)
+        {
+            value = v;
+        }
+
+        private Float16(bool sign, ushort exp, ushort sig) =>
```

Comment:
sig means mantissa?

Reply:
They call it significand. Most of Float16 class code is from System.Half class in a later version of the framework.